libs/SalesforceSDKCore/SalesforceSDKCore/Classes/PushNotification/SFPushNotificationManager.m

```diff
@@ -174,41 +171,31 @@ - (BOOL)registerForSalesforceNotifications
         [SFSDKCoreLogger e:[self class] format:@"Cannot register for notifications with Salesforce: no deviceToken"];
         return NO;
     }
-    NSMutableURLRequest *request = [[NSMutableURLRequest alloc] init];
-    
-    // URL and method
-    [request setURL:[NSURL URLWithString:kSFPushNotificationEndPoint relativeToURL:credentials.instanceUrl]];
-    [request setHTTPMethod:@"POST"];
-    
-    // Headers
-    [request setValue:@"application/json" forHTTPHeaderField:@"Content-Type"];
-    [request setValue:[NSString stringWithFormat:@"Bearer %@", credentials.accessToken] forHTTPHeaderField:@"Authorization"];
-    [request setHTTPShouldHandleCookies:NO];
-    
-    // Body
+    NSString *path = [NSString stringWithFormat:@"/%@/%@", [SFRestAPI sharedInstance].apiVersion, kSFPushNotificationEndPoint];
+    SFRestRequest *request = [SFRestRequest requestWithMethod:SFRestMethodPOST path:path queryParams:nil];
     NSString *bundleId = [NSBundle mainBundle].bundleIdentifier;
     NSDictionary* bodyDict = @{@"ConnectionToken":_deviceToken, @"ServiceType":@"Apple", @"ApplicationBundle":bundleId};
-    [request setHTTPBody:[SFJsonUtils JSONDataRepresentation:bodyDict]];
-    
-    // Send
-    SFNetwork *network = [[SFNetwork alloc] init];
-    [network sendRequest:request dataResponseBlock:^(NSData *data, NSURLResponse *response, NSError *error) {
-        if (error != nil) {
-            [SFSDKCoreLogger e:[self class] format:@"Registration for notifications with Salesforce failed with error %@", error];
+    [request setCustomRequestBodyDictionary:bodyDict contentType:@"application/json"];
```

Comment:
The code is essentially the same but refactored to use `SFRestAPI` instead of `SFNetwork`.